src/bacnet/basic/object/av.c

```diff
@@ -128,18 +128,68 @@ void Analog_Value_Init(void)
 }
 
 /**
- * We simply have 0-n object instances. Yours might be
- * more complex, and then you need validate that the
- * given instance exists.
+ * Initialize the analog values. Returns false if there are errors.
+ *
+ * @param pInit_data pointer to initialisation values
+ *
+ * @return true/false
+ */
+bool Analog_Value_Set(BACNET_OBJECT_LIST_INIT_T *pInit_data)
+{
+  unsigned i;
+
+  if (!pInit_data) {
+    return false;
+  }
+
+  if (pInit_data->length > MAX_ANALOG_VALUES) {
+    return false;
+  }
+
+  for (i = 0; i < pInit_data->length; i++) {
+    if (pInit_data->Object_Init_Values[i].Object_Instance < BACNET_MAX_INSTANCE) {
+      AV_Descr[i].Object_Instance = pInit_data->Object_Init_Values[i].Object_Instance;
```

Comment:
I was wondering... Why not to create an API function i.e. Analog_Value_Instance_Set(index, instance_no), export it via NIF and do the looping from Elixir side? Other funcs like Name_Set and Description_Set are already present....

Reply:
NIF calls are expensive, trying to pass in the data you need to the other side, and handle it there seemed better